### PR TITLE
Upgrade to terraform version 11.14. 

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

#2584 

## Purpose/Implementation Notes

I want to deploy this before jumping to 12.

It looks like Nomad recommends upgrading gradually. This one in particular could probably be skipped, but I want to take extra care.

This upgrade allows us to run a checklist for upgrading to terraform 12: https://www.terraform.io/upgrade-guides/0-12.html#terraform-0-12checklist

The checklist output:
```
After analyzing this configuration and working directory, we have identified some necessary steps that we recommend you take before upgrading to Terraform v0.12:

- [ ] Upgrade provider "aws" to version 3.12.0 or newer.
  
  No currently-installed version is compatible with Terraform 0.12. To upgrade, set the version constraint for this provider as follows and then run `terraform init`:
  
      version = "~> 3.12.0"

Taking these steps before upgrading to Terraform v0.12 will simplify the upgrade process by avoiding syntax errors and other compatibility problems.
```

I tried jumping right up to v3.12.0 for the AWS provider, but it said that it wasn't valid with the currently installed version of terraform. I'm not sure if this text is just set to specify the latest version of the AWS provider or if I need to upgrade the AWS provider at the same time as I upgrade terraform to version 12. (3.12.0 is the very latest version of the AWS provider, while version 12 is not the latest version of terraform.)

## Types of changes

- Upgrade

## Functional tests

I ran a dev deploy successfully.

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: infrastructure/instances.tf [hunk 1]
```diff
@@ -4,6 +4,8 @@
 data "aws_ami" "ubuntu" {
   most_recent = true
 
+  owners = ["589864003899"]
```
Comment (cgreene) [inline infrastructure/instances.tf:7]: couldn't tell how this was TF version related

File: infrastructure/networking.tf [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # related to networking.
 
 provider "aws" {
-  version = "1.37.0"
+  version = "2.70.0"
```
Comment (cgreene) [inline infrastructure/networking.tf:5]: also just check that this is intended

**Review by cgreene — APPROVED** [review]
Two things that weren't obviously terraform versions - added a note but feel free to merge if expected 👍

File: infrastructure/networking.tf (hunk 2 above)
Comment (kurtwheeler) [reply infrastructure/networking.tf:5]: This newer version of terraform didn't support this version of the AWS provider. `terraform 0.12-checklist` recommended version 3.12.0 which is the latest version, but that didn't support such an old version of terraform. This was the latest version starting with `2.` and `terraform 0.12-checklist` no longer recommended a later provider version.

**Pull request merged**